src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -179,8 +186,11 @@ private function hasProductWithPriceInChannel(array $products, int $price, Chann
                 continue;
             }
 
-            foreach ($product['variants'] as $variant) {
-                if ($variant['channelPricings'][$channel->getCode()]['price'] === $price) {
+            foreach ($product['variants'] as $variantIri) {
+                /** @var ProductVariantInterface $variant */
+                $variant = $this->iriConverter->getItemFromIri($variantIri);
+
+                if ($variant->getChannelPricingForChannel($channel)->getPrice() === $price) {
```

Comment:
This check doesn't check the API behaviour in 100%, you should probably get channel pricing as another call to API or add channel pricings to variant (but it would be a more temporary solution as we don't want to expose all channel pricings)